### DISCUSSION TITLE
Using one QueryClient instance in multiple QueryClientProviders

### DISCUSSION THREAD

**Discussion by kmiwa007** [TanStack/query#4908] [Q&A]
Hello,

I'm working on migrating an app from AngularJS to React. There are multiple entry points of new React components into the existing Angular app. We are using ReactQuery for server state management.  Each top-level React component inserted into the component hierarchy will be wrapped with `<QueryClientProvider>`.  If I want all components to use the same ReactQuery cache no matter where they are inserted, does it make sense to have one QueryClient instance which is passed into multiple QueryClientProviders?

**Comment by TkDodo** [comment]
sure :)
